core/provider/service.go

```diff
@@ -610,42 +610,31 @@ func (s *Service) fetchNewResources(ctx context.Context, p *domain.Provider) ([]
 	}
 	flattenedProviderResources := flattenResources(filteredResources)
 
-	existingProviderResources := map[string]bool{}
+	// convert existing resources into map[string(GlobalURN)]*domain.Resource
+	// recursively compare newResources with existingResources
+	// -> if a particular resource is not updated (diff = 0), then, remove this current resource from the parent's.Children
+	// -> 2nd action, mark the resource in existingResourcesMap as deleted or not
+	// -> at the end of recursive evaluation, we will get list of resources that are updated and the hierarcy is still maintained
+	//
+	// get all resources from existingResourcesMap that are
+	mapExistingResources := make(map[string]*domain.Resource)
+	for _, existing := range existingGuardianResources {
+		mapExistingResources[existing.GlobalURN] = existing
+	}
 	updatedResources := []*domain.Resource{}
+	existingProviderResources := map[string]bool{}
+
 	for _, newResource := range flattenedProviderResources {
```

Comment:
we should no longer use the flattened resources and process `newProviderResources` directly. we can create a dummy parent resource and pass that dummy parent to the recursive function
```go
dummyParent := &domain.Resource{
  // ...
  Children: newProviderResources,
}

res := s.diffResources(ctx, mapExistingResources, dummyParent)

// continue upserting res.Children to db
```